server/graphql/common/expenses.ts

```diff
@@ -300,6 +301,16 @@ export const canApprove: ExpensePermissionEvaluator = async (req, expense, optio
     }
     return false;
   } else {
+    const collective = expense.collective || (await models.Collective.findByPk(expense.CollectiveId));
+    if (collective.hasPolicy(POLICIES.EXPENSE_AUTHOR_CANNOT_APPROVE) && req.remoteUser.id === expense.UserId) {
```

Comment:
We must try to stick with loaders as much as we can in these permissions evaluators. This one is called in the expenses list, a loader will prevent calling 15x the DB

```suggestion
    expense.collective = expense.collective || (await req.loaders.Collective.byId.load(expense.CollectiveId));
    if (expense.collective?.hasPolicy(POLICIES.EXPENSE_AUTHOR_CANNOT_APPROVE) && req.remoteUser.id === expense.UserId) {
```

Reply:
Perfect! I feel like I always forget about loaders and you're always catching that. 😂